### PR TITLE
editoast: switch to diesel_async

### LINKED ISSUE #4829: Editoast: switch to diesel_async

### PR DESCRIPTION
**Pull request by Tristramg**
Closes #4829

### CODE REVIEW

File: editoast/editoast_derive/src/infra_model.rs [hunk 1]
```diff
@@ -39,11 +39,13 @@ pub fn infra_model(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
                 // Work around a diesel limitation
                 // See https://github.com/diesel-rs/diesel/issues/2414
                 // Divided by 3 here because we are inserting three values
-                const DIESEL_MAX_VALUES : usize = (2_usize.pow(16) - 1)/3;
+                // When using AsyncPgConnection, we divide must again
```
Comment (leovalais) [inline editoast/editoast_derive/src/infra_model.rs]:
```suggestion
                // When using AsyncPgConnection, we must divide again
```

File: editoast/src/views/infra/mod.rs [hunk 2]
```diff
@@ -273,26 +271,25 @@ async fn clone(
...
 ) -> Result<Json<i64>> {
-    let db_pool_ref = db_pool.clone();
     let mut futures = Vec::<Pin<Box<dyn Future<Output = _>>>>::new();
 
     let infra = infra.into_inner();
     let name = new_name.name.clone();
-    let cloned_infra = Infra::clone(infra, db_pool_ref.clone(), name).await?;
-
+    let cloned_infra = Infra::clone(infra, db_pool.clone(), name).await?;
+    // When creating a connection for each objet, it will a panic with 'Cannot access shared transaction state' in bb8 database pool
```
Comment (leovalais) [inline editoast/src/views/infra/mod.rs]: Could changing the database pool to either Deadpool or mobc fix the problem?

**Review by leovalais — COMMENTED** [review]
That's a huge amount of work! It's nice to have everything decluttered by those blocks :) Thanks a lot!

I've just read the code for now and left a few cosmetic suggestions.
I wish the `async move { ... }.scope_boxed()` pattern could be avoided, too bad we can't use async closures. Maybe a macro to hide the boilerplate would work?

In any case, even of the CI says it's good, we'll have to thoroughly test the app to make sure there's no hidden panic or weird behaviors introduced. I'll do that next week (I'm not against some help with that @Tguisnet @flomonster 😛).

File: editoast/src/views/infra/mod.rs (hunk 2 above)
Comment (Tristramg) [reply editoast/src/views/infra/mod.rs]: I tried with deadpool, same problem. Do you have an opinion between those?

The API is almost the same, but creating the pool is sync, with makes a smaller changeset

**Review by flomonster — APPROVED** [review]
Thanks!
Reviewed and tested. All features seems to work.

**Review by leovalais — APPROVED** [review]

**Pull request merged**